src/Contexts/CartContext.js

```diff
@@ -1,35 +1,90 @@
-import React ,{createContext,useContext,useReducer} from "react"
-
-
-const CartContext = createContext()
-
-function reducerfn(previousState,action)
-{
-    switch(action.type)
-    {
-        case "ADD_TO_CART":
-            return {...previousState,data:[...previousState.data,...action.payload.data]}
-        case "REMOVE_FROM_CART":
-            return {...previousState}
-        
-        case "ADD_TO_WISHLIST":
-            return {...previousState,wishList:[...previousState.wishList,...action.payload.data]}
-        
-        default:
-            return {...previousState}
-    }
-}
 
+import React, { createContext, useContext, useReducer } from 'react';
+
+const CartContext = createContext();
+
+function reducerfn(previousState, action) {
+  switch (action.type) {
+    case 'ADD_TO_CART':
+      return {
+        ...previousState,
+        data: [...previousState.data, ...action.payload.data],
+      };
+
+    case 'ADD_TO_WISHLIST':
+      return {
+        ...previousState,
+        wishList: [...previousState.wishList, ...action.payload.data],
+      };
+    case 'MOVE_TO_WISHLIST':
+      const wishListdata = previousState.data.filter(
+        (element) => element.id !== action.payload.id
+      );
+      const cartData = previousState.data.filter(
+        (element) => element.id === action.payload.id
+      );
+      return {
+        ...previousState,
+        wishList: [...previousState.wishList, wishListdata],
+        data: [...cartData],
+      };
+    case 'MOVE_TO_CART':
+      const cartDataFromWishList = previousState.wishList.filter(
```

Comment:
Suggestion: create a new file for reducer

Reply:
will do